### ISSUE TITLE
TypeError & Exception with spades/MaxBin2 files

### ISSUE THREAD

**Issue by michaelwoodworth** [Vini2/GraphBin#9]
Hi GraphBin group, I was able to run GraphBin and get output that looks correct for the majority of my files.  I have a subset of 8 of my 45 files that are all getting the same errors.  I have double checked the content of these files, which seem to be fine.  I'm copying the code & output below, would you let me know if there are any workarounds you might suggest? Thanks!

python ${path}/GraphBin/graphbin.py --assembler spades --graph ${spades_dir}/${name}/assembly_graph_with_scaffolds.gfa --paths ${spades_dir}/${name}/contigs.paths --binned ${path}/graphbin/inputs/CSVs/${name}_initial_contig_bins.csv --output ${outdir}/${name}

2020-07-23 13:12:22,545 - INFO - Welcome to GraphBin: Refined Binning of Metagenomic Contigs using Assembly Graphs.
2020-07-23 13:12:22,547 - INFO - This version of GraphBin makes use of the assembly graph produced by SPAdes which is based on the de Bruijn graph approach.
2020-07-23 13:12:22,547 - INFO - Input arguments:
2020-07-23 13:12:22,547 - INFO - Assembly graph file: ${path}/PM2-C1D1/assembly_graph_with_scaffolds.gfa
2020-07-23 13:12:22,547 - INFO - Contig paths file: ${path}/PM2-C1D1/contigs.paths
2020-07-23 13:12:22,547 - INFO - Existing binning output file: ${path}/inputs/CSVs/PM2-C1D1_initial_contig_bins.csv
2020-07-23 13:12:22,547 - INFO - Final binning output file: ${path}/gb_bins/PM2-C1D1/
2020-07-23 13:12:22,547 - INFO - Maximum number of iterations: 100
2020-07-23 13:12:22,547 - INFO - Difference threshold: 0.1
2020-07-23 13:12:22,547 - INFO - GraphBin started
2020-07-23 13:12:22,567 - INFO - Number of bins available in the initial binning result: 14
2020-07-23 13:12:22,567 - INFO - Constructing the assembly graph
2020-07-23 13:12:23,228 - INFO - Total number of contigs available: 60554
2020-07-23 13:12:28,439 - INFO - Total number of edges in the assembly graph: 7011
2020-07-23 13:12:28,439 - INFO - Obtaining the initial binning result
2020-07-23 13:12:28,452 - INFO - Determining ambiguous vertices
2020-07-23 13:12:28,936 - INFO - Removing labels of ambiguous vertices
2020-07-23 13:12:28,988 - INFO - Obtaining the refined binning result
2020-07-23 13:12:28,988 - INFO - Deteremining vertices which are not isolated and not in components without any labels
2020-07-23 13:12:36,162 - INFO - Number of non-isolated contigs: 4678
Traceback (most recent call last):
  File "${path}/GraphBin/src/labelpropagation/labelprop.py", line 113, in process_data_line
    for edge in edges:
TypeError: 'int' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "${path}/GraphBin/src/graphbin_SPAdes.py", line 497, in <module>
    lp.load_data_from_mem(data)
  File "${path}/GraphBin/src/labelpropagation/labelprop.py", line 99, in load_data_from_mem
    self.process_data_line(line)
  File "${path}/apps/GraphBin/src/labelpropagation/labelprop.py", line 121, in process_data_line
    raise Exception("Coundn't parse vertex from line")
Exception: Coundn't parse vertex from line

**Comment by anuradhawick** [comment]
Thanks for raising the issue.

Could you please share the error causing files if that's possible? This might have caused by an edge case we have missed or a slightly different naming convention in the file. You can reach us on anuradha.wickramarachchi@anu.edu.au or vijini.mallawaarachchi@anu.edu.au.

**Comment by Vini2** [comment]
This issue was created as the initial binning result consisted of contigs that belonged to multiple bins. GraphBin is designed to handle initial contigs which belong to only one bin. Multiple bins for the initial contigs are not supported. We have added a check to verify that each contig in the initial binning result should belong to only one bin. 

Fixed in commit ID [7128e19938a421d64324665c8477ea1ec3bae8b4](https://github.com/Vini2/GraphBin/commit/7128e19938a421d64324665c8477ea1ec3bae8b4).

Closing this issue after fixing.
